hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/StripeReader.java

```diff
@@ -284,7 +307,8 @@ private Callable<BlockReadStats> readCells(final BlockReader reader,
 
       int ret = 0;
       for (ByteBufferStrategy strategy : strategies) {
-        int bytesReead = readToBuffer(reader, datanode, strategy, currentBlock);
+        int bytesReead = readToBuffer(reader, datanode, strategy, currentBlock,
+            chunkIndex);
```

Comment:
For `readToBuffer` maybe need to consider the current actual offsetInBlock.

`readToBuffer(reader, datanode, strategy, currentBlock, chunkIndex, ret);`